tool/tsh/proxy.go

```diff
@@ -223,13 +239,42 @@ func sshProxy(tc *libclient.TeleportClient, params sshProxyParams) error {
 		args = append([]string{"-l", tc.HostLogin}, args...)
 	}
 
+	if logrus.GetLevel() == logrus.DebugLevel {
+		args = append([]string{"-v"}, args...)
+	}
+
 	log.Debugf("Executing proxy command: %v %v.", sshPath, strings.Join(args, " "))
 
 	child := exec.Command(sshPath, args...)
 	child.Stdin = os.Stdin
 	child.Stdout = os.Stdout
-	child.Stderr = os.Stderr
-	return trace.Wrap(child.Run())
+
+	// Since we want to capture the stderr in case of an error,
+	// create a stderr pipe and tee it into os.stderr after reading it
+	stdErrPipe, err := child.StderrPipe()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	defer stdErrPipe.Close()
+
+	if err := child.Start(); err != nil {
+		return trace.Wrap(err)
+	}
+
+	stdErr, err := io.ReadAll(io.TeeReader(stdErrPipe, os.Stderr))
```

Comment:
In successful flow we are keeping internal buffer for stderr 
Can we consider using similar approach to [PR](https://github.com/gravitational/teleport/pull/13843/files#diff-dae87d4d3799335d85a5fa6ff17cfcbfb30705ae145c9e67ef70286423676a48R670-R671) 
where we only read `n` first byte in order to check the error ?

Reply:
Hmm I'm not sure I agree, https://github.com/gravitational/teleport/pull/13843#discussion_r907690842 seems like an edge case. I think in the normal case of running an exec command it makes sense to capture the full stderr. In this case, returing any stderr is useful both for error logging and for potential error matching.

I do like the option of using an `io.MultiWriter` instead of Pipe+Tee though.